### PR TITLE
Documentation for Collateral related items

### PR DESCRIPTION
**Pull request by ajsantander**

### CODE REVIEW

File: packages/synthetix-main/contracts/storage/Collateral.sol [hunk 1]
```diff
@@ -6,29 +6,72 @@ import "@synthetixio/core-contracts/contracts/utils/SetUtil.sol";
...
 import "./Pool.sol";
 
+/**
+ * @title Stores information about a deposited asset for a given account.
+ *
+ * This is not a global data structure, but rather one that will have an entry for each account > asset pair.
+ *
+ * That is, accounts have one of these data structures for each type of collateral they utilize.
+ *
+ * TODO: Consider renaming to CollateralEntry to indicate that this is just an entry, rather than a global thing.
```
Comment (sunnyvempati) [inline packages/synthetix-main/contracts/storage/Collateral.sol:16]: yes to this.

File: packages/core-contracts/contracts/interfaces/IERC20.sol [hunk 2]
```diff
@@ -2,6 +2,9 @@
 pragma solidity ^0.8.0;
 
 interface IERC20 {
+    error InsufficientAllowance(uint required, uint existing);
+    error InsufficientBalance(uint required, uint existing);
```
Comment (sunnyvempati) [inline packages/core-contracts/contracts/interfaces/IERC20.sol:6]: should events also be in interfaces?  not sure whats the standard here and the reasons behind it

**Review by sunnyvempati — APPROVED** [review]

File: packages/synthetix-main/contracts/storage/Collateral.sol (hunk 1 above)
Comment (noahlitvin) [reply packages/synthetix-main/contracts/storage/Collateral.sol:16]: Maybe `DepositedCollateral`?

File: packages/core-contracts/contracts/interfaces/IERC20.sol (hunk 2 above)
Comment (ajsantander) [reply packages/core-contracts/contracts/interfaces/IERC20.sol:6]: Yes I think so. Events in interfaces should help subgraph developers.

**Review by noahlitvin — APPROVED** [review]

**Pull request merged**